ambry-store/src/test/java/com.github.ambry.store/CompactionLogTest.java

```diff
@@ -0,0 +1,296 @@
+  }
+
+  // Variables that represent the folder where the data resides
+  private final File tempDir;
+  private final String tempDirStr;
+  // the time instance that will be used in the index
+  private final Time time = new MockTime();
+
+  /**
+   * Creates a temporary directory for the compaction log file.
+   * @throws IOException
+   */
+  public CompactionLogTest() throws IOException {
+    tempDir = StoreTestUtils.createTempDirectory("storeDir-" + UtilsTest.getRandomString(10));
+    tempDirStr = tempDir.getAbsolutePath();
+  }
+
+  /**
+   * Deletes the temporary directory.
+   * @throws InterruptedException
+   * @throws IOException
+   */
+  @After
+  public void cleanup() throws InterruptedException, IOException, StoreException {
+    assertTrue(tempDir.getAbsolutePath() + " could not be deleted", StoreTestUtils.cleanDirectory(tempDir, true));
+  }
+
+  /**
+   * Tests the use of {@link CompactionLog} when it is used without closing b/w operations and compaction cycles.
+   * @throws IOException
+   */
+  @Test
+  public void iterationWithoutReloadTest() throws IOException {
+    String storeName = "store";
+    List<CompactionDetails> detailsList = getCompactionDetails(5);
+    assertFalse("Compaction should not be in progress", CompactionLog.isCompactionInProgress(tempDirStr, storeName));
+    CompactionLog log = new CompactionLog(tempDirStr, storeName, time, detailsList);
+    assertTrue("Compaction should should be in progress", CompactionLog.isCompactionInProgress(tempDirStr, storeName));
+    for (CompactionDetails details : detailsList) {
+      verifyEquality(details, log.getCompactionDetails());
+      assertEquals("Should be in the PREPARE phase", CompactionLog.Phase.PREPARE, log.getCompactionPhase());
+      log.markCopyStart();
+      assertEquals("Should be in the COPY phase", CompactionLog.Phase.COPY, log.getCompactionPhase());
+      StoreFindToken safeToken =
+          new StoreFindToken(new MockId("dummy"), new Offset(LogSegmentNameHelper.generateFirstSegmentName(5), 0),
+              new UUID(1, 1));
+      log.setSafeToken(safeToken);
+      assertEquals("Returned token not the same as the one that was set", safeToken, log.getSafeToken());
+      log.markSwitchStart();
+      assertEquals("Should be in the SWITCH phase", CompactionLog.Phase.SWITCH, log.getCompactionPhase());
+      log.markCleanupStart();
+      assertEquals("Should be in the CLEANUP phase", CompactionLog.Phase.CLEANUP, log.getCompactionPhase());
+      log.markCycleComplete();
+    }
+    assertEquals("Should be in the DONE phase", CompactionLog.Phase.DONE, log.getCompactionPhase());
+    log.close();
+    assertFalse("Compaction should not be in progress", CompactionLog.isCompactionInProgress(tempDirStr, storeName));
+  }
+
+  /**
+   * Tests the use of {@link CompactionLog} when it is closed and reloaded b/w operations and compaction cycles.
+   * @throws IOException
+   */
+  @Test
+  public void iterationWithReloadTest() throws IOException {
```

Comment:
Good point. I considered that but it has a lot of `if` blocks and turns out that the amount of code is quite similar. Because of that, I decided separating them would be more readable and clear.